connector/datadogconnector/connector.go

```diff
@@ -113,6 +130,23 @@ func (c *traceToMetricConnector) Capabilities() consumer.Capabilities {
 
 func (c *traceToMetricConnector) ConsumeTraces(ctx context.Context, traces ptrace.Traces) error {
 	c.agent.Ingest(ctx, traces)
+	if c.enableContainerStats && len(c.containerTagAttrs) > 0 {
```

Comment:
Should this happen before the `c.agent.Ingest` call in case of possible race conditions?